### PR TITLE
Prebid core: do not use `document.write` for rendering HTML

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Refactoring (no functional changes, no api changes)

## Description of change

Replace core's usage of `document.write` with `Element.innerHTML`, with some additional logic to run scripts.

According to:

https://www.w3.org/TR/2008/WD-html5-20080610/dom.html#innerhtml0
https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML

'script' is the only major difference called out between  `document.write` and `innerHTML`, with the rest being about state of the browser's internal parsers - which should hopefully not matter for us.

## Other information

Closes https://github.com/prebid/Prebid.js/issues/8337

### CODE REVIEW

File: src/utils/writeHTML.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
+ * Replace the contents of `element` with the given `html` string, running all scripts contained within it.
+ *
+ * @param element the container element. Must be in the DOM.
+ * @param html HTML string
+ */
+export function replaceHTML(element, html) {
```
Comment (mkendall07) [inline src/utils/writeHTML.js:7]: My only concern with this approach is that I believe it makes scripts execute asynchronously, where as using `document.write` will execute synchronous. So you might hit some edge cases here.

**Comment by dgirardi** [comment]
As it turns out from the test failures, it's more complicated than even the sync/async problem. I don't think this approach will work - but maybe someone has a better idea. We could try to pull in postscribe, but I'm not sure it's worth it.

In Firefox, newly created iframes behave like this:

```
 const iframe = document.createElement('iframe');
 document.appendChild(iframe);
 const doc = iframe.contentDocument;
 doc.body.appendChild(...) // does not work as expected, because:
 iframe.contentWindow.addEventListener('DOMContentLoaded', function() {
    doc === iframe.contentDocument // this is false in Firefox
    doc.defaultView.document === iframe.contentDocument // but this, somehow, is true
 })
```

I spent some time to try and code that in this PR, but I am running into other problems - I think because  scripts do not run at the expected time.

**Comment by patmmccann** [comment]
Should core import postscribe?

**Comment by nikpil93** [comment]
> As it turns out from the test failures, it's more complicated than even the sync/async problem. I don't think this approach will work - but maybe someone has a better idea. We could try to pull in postscribe, but I'm not sure it's worth it.
> 
> In Firefox, newly created iframes behave like this:
> 
> ```
>  const iframe = document.createElement('iframe');
>  document.body.appendChild(iframe);
>  const doc = iframe.contentDocument;
>  doc.body.appendChild(...) // does not work as expected, because:
>  iframe.contentWindow.addEventListener('DOMContentLoaded', function() {
>     doc === iframe.contentDocument // this is false in Firefox
>     doc.defaultView.document === iframe.contentDocument // but this, somehow, is true
>  })
> ```
> 
> I spent some time to try and code that in this PR, but I am running into other problems - I think because scripts do not run at the expected time.

Hello, couldn’t you use the srcdoc attribute of the iframe (srcdoc=“html code”) in order to apply the htmlcontent accordingly? It seems that low effort is needed and you could combine the src=“javascript:false” to avoid the default iframe and DOM behavior. Am I missing something?
CC: @dgirardi @mkendall07 @patmmccann 
https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/srcdoc
